### DISCUSSION TITLE
Hacker house challenges

### DISCUSSION THREAD

**Discussion by AbdelStark** [keep-starknet-strange/madara#864] [Ideas]
# Build the first POW Starknet Appchain

Resources:
- https://docs.substrate.io/reference/how-to-guides/consensus-models/add-proof-of-work-consensus/
- https://github.com/substrate-developer-hub/substrate-how-to-guides/tree/main/example-code/consensus-nodes/POW

**Comment by AbdelStark** [comment]
# Madara config generator

Resources:
- https://paritytech.github.io/parity-config-generator/#config=eyJfX2ludGVybmFsIjp7ImNvbmZpZ01vZGUiOiJhZHZhbmNlZCJ9LCJwYXJpdHkiOnsiY2hhaW4iOiJrb3ZhbiJ9LCJycGMiOnsicG9ydCI6OTk0NH19

**Comment by antiyro** [comment]
# Madara TUI real-time dashboard

### Project Overview:

The idea would be to implement a TUI dashboard using [ratatui.rs](https://ratatui.rs/) that will serve as a visual interface for node operators, displaying critical information during the node syncing process and regular operations. This tool aims to make node management more intuitive and accessible, especially for those preferring terminal-based interactions.

![3rel2b3m8iw81](https://github.com/keep-starknet-strange/madara/assets/74653697/4c6a7ff8-298f-4731-900f-f2153f60bc7b)

### Core Features:

1. **Sync Progress**: Display real-time sync status, including current block height, percentage completion, and a visual progress indicator.
2. **Network Stats**: Show live network information such as number of peers, bandwidth usage, and other relevant network metrics.
3. **Resource Utilization**: Monitor the node's CPU, memory, and disk usage to help operators understand the resource impact of their node.
4. **Consensus Details**: Provide information on the consensus mechanism, including validator status and finalized block count.
5. **Log Stream**: Integrate a live feed of logs, highlighting errors and warnings for quick troubleshooting.
6. **Customization Options**: Allow users to tailor the dashboard layout and displayed metrics according to their preferences.

### Resources

https://ratatui.rs/
